horovod/run/gloo_run.py

```diff
@@ -190,9 +192,15 @@ def set_event_on_sigterm(signum, frame):
     # a SIGINT, the event will be set and the spawned threads will kill their
     # corresponding middleman processes and thus the jobs will be killed as
     # well.
-    threads.execute_function_multithreaded(_exec_command,
-                                           args_list,
-                                           block_until_all_done=True)
+    res = threads.execute_function_multithreaded(_exec_command,
+                                                 args_list,
+                                                 block_until_all_done=True)
+    for name in res:
+        if res[name] != 0:
+            raise RuntimeError("Gloo job detected that one or more processes exited with non-zero "
```

Comment:
Because `res` is an unordered dict, we cannot know the order that the processes terminated, so it's not accurate to say that the first process to exit was <name>.  So, here's what I'd suggest doing instead:

In `_exec_command`, instead of returning a single integer exit code, try returning a tuple: `(exit_code, time.time())`.  That will give you a floating-point timestamp, which you can then use to sort the dictionary like so:

```
def get_timestamp(item):
    name, value = item
    exit_code, timestamp = value
    return timestamp

for name, value in sorted(res.items(), key=get_timestamp):
    exit_code, timestamp = value
    ...
```